R/metab_model-class.R

```diff
@@ -0,0 +1,120 @@
+#### class definition ####
+
+#' A generic metabolism model class.
+#' 
+#' Class and function definitions for a metabolism model
+#' 
+#' @rdname metab_model-class
+#' @name metab_model-class
+#' @slot fit An internal representation of a fitted model.
+#' @slot args A list of arguments, excluding data, that were supplied to the fitting function.
+#' @slot data The data that were used to fit the model.
+#' @exportClass metab_model
+#' @family metab.model.classes
+setClass(
+  "metab_model",
+  slots=c(
+    fit="ANY",
```

Comment:
the fit arg is meant to contain an internal representation of any type, abstracted away from the user but known to the developers, so that different model types can be stored in whatever format is most natural to them while still being available to the user via an interface that's the same for every model type.